### PR TITLE
Cleaner way to detect if we can clone via ssh or not

### PR DESCRIPTION
**Pull request by grimpy**
Add dependency requirements and point to the docs

See: #817, #818

Signed-off-by: Jo De Boeck <deboeck.jo@gmail.com>

### CODE REVIEW

File: install/threesdk/InstallTools.py [hunk 1]
```diff
@@ -3417,51 +3441,34 @@ def assert_merge(args, repo_url):
                         showout=True,
                         interactive=True,
                         retry=4,
-                        errormsg="Could not clone %s" % repo_url,
+                        errormsg=f"Could not clone {url}",
```
Comment (waleedhammam) [inline install/threesdk/InstallTools.py:3444]: Shouldn't be `URL` to be parsed from args ?

File: install/threesdk/InstallTools.py [hunk 2]
```diff
@@ -3417,51 +3441,34 @@ def assert_merge(args, repo_url):
...
+                    Tools.log("get code [https] (default branch): %s" % repo)
+                    C = "git -C {ACCOUNT_DIR} clone -q {URL}"
+                    Tools.execute(
+                        C,
+                        args=args,
+                        die=True,
+                        showout=True,
+                        interactive=True,
+                        retry=4,
+                        errormsg=f"Could not clone {url}",
```
Comment (waleedhammam) [inline install/threesdk/InstallTools.py:3458]: also Shouldn't be `URL` to be parsed from args ?

File: install/threesdk/InstallTools.py [hunk 3]
```diff
@@ -3487,11 +3494,11 @@ def assert_merge(args, repo_url):
                                     raise Tools.exceptions.Input("found changes, do not want to commit or reset")
                     # update repo
                     Tools.execute(
-                        "git -C {REPO_DIR} fetch -q",
+                        "git -C {REPO_DIR} fetch -q {URL}",
                         args=args,
                         retry=4,
                         showout=False,
-                        errormsg="Could not pull %s" % repo_url,
+                        errormsg=f"Could not pull {url}",
```
Comment (waleedhammam) [inline install/threesdk/InstallTools.py:3501]: same like Shouldn't be `URL` to be parsed from args ?

File: install/threesdk/InstallTools.py [hunk 4]
```diff
@@ -3504,7 +3511,7 @@ def assert_merge(args, repo_url):
...
```
Comment (waleedhammam) [inline install/threesdk/InstallTools.py:3514]: same

File: install/threesdk/InstallTools.py [hunk 5]
```diff
@@ -59,6 +60,10 @@
     pygments_formatter = False
     pygments_pylexer = False
 
+RepoInfo = namedtuple("RepoInfo", ["protocol", "host", "account", "name", "url", "port"])
+
+GITHUB_RSA = "AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ=="
```
Comment (waleedhammam) [inline install/threesdk/InstallTools.py:65]: is it ok to hardcode  GITHUB_RSA  ? permission or what if i need my key ? or just for known hosts

File: install/threesdk/InstallTools.py [hunk 6]
```diff
@@ -3308,31 +3344,26 @@ def checkoutbranch(args, branch):
...
             cd {REPO_DIR}
-            git pull -q
+            git pull -q {URL}
             """
             rc, out, err = Tools.execute(
                 C,
                 args=args,
                 retry=1,
-                errormsg="Couldn't merge repo %s" % repo_url,
+                errormsg=f"Couldn't merge repo {args['url']}",
```
Comment (waleedhammam) [inline install/threesdk/InstallTools.py:3366]: Shouldn't be URL to be parsed from args ?

**Review by waleedhammam — APPROVED** [review]

**Review by rkhamis — APPROVED** [review]
but don't merge to development. merge to rc branch (from master)

**Comment by grimpy** [comment]
Merged on the rc branch instead

**Pull request closed without merging**